aws/resource_aws_guardduty_member.go

```diff
@@ -93,10 +143,12 @@ func resourceAwsGuardDutyMemberRead(d *schema.ResourceData, meta interface{}) er
 		d.SetId("")
 		return nil
 	}
+
 	member := gmo.Members[0]
 	d.Set("account_id", member.AccountId)
 	d.Set("detector_id", detectorID)
 	d.Set("email", member.Email)
+	d.Set("relationship_status", member.RelationshipStatus)
 
```

Comment:
It might be worth trying to correctly `d.Set("invite", ...)` based on the value of `member.RelationshipStatus` -- this will allow for a way to upgrade existing configurations. These seem like they would indicate the account was invited at one point: `INVITED`, `DISABLED`, `ENABLED`

Reply:
I'm saying we should probably try to detect if an invite has already been sent based on the relationship status, which would be done correctly setting true/false into `d.Set("invite", XXX)` during the read function.

I imagine most people will have already sent member invites (outside of Terraform) with the existing `aws_guardduty_member` resource. After upgrading their AWS provider with this new code and adding `invite = true` or `invite = false` into existing Terraform configurations, Terraform will currently try to recreate those GuardDuty member as the `invite` attribute is `ForceNew: true`. It'll showing a plan of `"" => "true" (forces new resource)` or `"" => "false" (forces new resource)`.  We can add `d.Set("invite", XXX)` during the read function so Terraform knows there is not a difference and will not try to recreate the resource.

Reply:
hmmm, ok let me see